### PR TITLE
Revert install-w-machine.md to 833471b74cf39c826f349c0355dfca0d244f92c0.

### LINKED ISSUE #2441: Failed to follow the doc `Evaluate Swarm in a sandbox` to create a swarm
Issue was opened at https://github.com/docker/docker/issues/24982. But it should be resolve at Swarm. 

cc @twang2218 @sblackstone.

### PR DESCRIPTION
**Pull request by dongluochen**
Fix #2441. 

Signed-off-by: Dong Chen <dongluo.chen@docker.com>

### CODE REVIEW

**Comment by dongluochen** [comment]
I don't remember why `install-w-machine.md` was changed this way but it's broken. I tested the previous one using `--swarm` and it works fine. Talked to @nathanleclaire and that should work (otherwise a bug). 

Cc @nishanttotla, @allencloud, @SvenDowideit

**Comment by nishanttotla** [comment]
@dongluochen I remember that the reasoning was to make it more transparent wrt the creation of the swarm, rather than rely on machine magic. The previous version (the one you're restoring to) did work, I can confirm too.

**Comment by dongluochen** [comment]
Thanks @nishanttotla. I think it was trying to translate `docker-machine` integration to simple and more steps. But it's easier to get wrong without thorough testing. I think we can revert it for now.

**Comment by nathanleclaire** [comment]
It's true, the motivation for the original change was to make the internals more transparent.

Have you debugged why / where the current doc is failing?  And did you check to see if it works with 0.7.0 as well?  Like I said in chat IIRC this doc has been kind of busted for a while. It might just need a little love.

**Comment by dongluochen** [comment]
@nathanleclaire With the steps in the current doc, swarm manager cannot access agents thru TLS. The certificates are not compatible with each other. 

```
DEBU[0060] Failed to validate pending node: The server probably has client authentication (--tlsverify) enabled. Please check your TLS client certification settings: Get https://192.168.99.101:2376/info: remote error: bad certificate  Addr=192.168.99.101:2376
```

**Comment by nathanleclaire** [comment]
@dongluochen Did you try with a fresh store (`-s` global option for machine)?  There may be existing cert "cache" which is causing issue due to 0.8.0 TLS changes

**Comment by nathanleclaire** [comment]
The `--swarm` commands work for me w/ fresh store so it's possible to get it to work

**Comment by dongluochen** [comment]
I feel like `--swarm` is a nice feature to create swarm with docker-machine. It's the simplest way of setting up Swarm which provides great user experience. The [swarm at scale doc](https://docs.docker.com/swarm/swarm_at_scale/) provides details on running swarm in production. 

Ping @SvenDowideit on documentation test failure.

**Comment by twang2218** [comment]
@nathanleclaire I tried `-s` options, it's still not working.

I modified my original script from  docker/docker#24982 to use the `-s` option, (Actually, I use the `MACHINE_STORAGE_PATH` environment variable to simplified the code). And I have put the script, output, `docker info`, and logs for both swarm manager and agent on the gist <https://gist.github.com/twang2218/55528df524f75cf9d941ad6c0327b396>

The Swarm manager still keep complaining the following in the log:

```bash
DEBU[1270] Failed to validate pending node: The server probably has client authentication (--tlsverify) enabled. Please check your TLS client certification settings: Get https://192.168.99.101:2376/info: remote error: bad certificate  Addr=192.168.99.101:2376
DEBU[1270] Failed to validate pending node: The server probably has client authentication (--tlsverify) enabled. Please check your TLS client certification settings: Get https://192.168.99.102:2376/info: remote error: bad certificate  Addr=192.168.99.102:2376
```

**Comment by dongluochen** [comment]
Ping @SvenDowideit. We want to add this change to next release along with Docker 1.12.1.

File: docs/install-w-machine.md [hunk 1]
```diff
@@ -1,194 +1,236 @@
...
+		fe0cc96a72cf04dba8c1c4aa79536ec3
 
-## Create a Swarm discovery token
+	The `swarm create` command returned the  `fe0cc96a72cf04dba8c1c4aa79536ec3`
+	token.
 
-Here you use the discovery backend hosted on Docker Hub to create a unique discovery token for your cluster. This discovery backend is only for low-volume development and testing purposes, not for production. Later on, when you run the Swarm manager and nodes, they register with the discovery backend as members of the cluster that's associated with the unique token. The discovery backend maintains an up-to-date list of cluster members and shares that list with the Swarm manager. The Swarm manager uses this list to assign tasks to the nodes.
+	**Note**: This command relies on Docker Swarm's hosted discovery service. If
+	this service is having issues, this command may fail. In this case, see
+	information on using other types of [discovery backends](./discovery). Check
```
Comment (SvenDowideit) [inline docs/install-w-machine.md]: This is the link that the checker is (correctly) telling you is not right.

```
01:39:54 ERROR: (290) 1 links to (swarm/discovery)
01:39:54 			 link ./discovery on page swarm/install-w-machine.md
```

**Comment by dongluochen** [comment]
Thank you @SvenDowideit! I didn't know I should check the console output 😔

**Comment by nishanttotla** [comment]
LGTM

**Comment by twang2218** [comment]
Then, what about original approach? Using `docker-machine` to create several simple virtualbox docker hosts, and use `docker run swarm` to create/join swarm manually?

This `docker-machine --swarm` did the trick to create a swarm easily, but without the manual approach, people cannot try to manually create the swarm simply at local host, especially for the people don't use `docker-machine` to build swarm.

Yes, the `install-manual.md` did describe how to create the swarm manually, but it's on AWS cloud, which is not as easy as local virtualbox machines.

I still think it's a problem should be fixed, rather than use `--swarm` as a workaround.

**Pull request merged**